src/name_analysis.rs

```diff
@@ -164,30 +164,6 @@ pub struct TargetNode {
     pub info: NodeInfo
 }
 
-#[derive(Debug, PartialEq)]
-pub struct Type {
-    pub name: String,
-    pub options: Vec<TypeOptionNode>
-}
-
-#[derive(Debug, PartialEq)]
-pub struct TypeNode {
-    pub val: Type,
-    pub info: NodeInfo
-}
-
-#[derive(Debug, PartialEq)]
-pub struct TypeOption {
-    pub name: String,
-    pub args: Vec<String>
-}
-
-#[derive(Debug, PartialEq)]
-pub struct TypeOptionNode {
-    pub val: TypeOption,
-    pub info: NodeInfo
-}
-
```

Comment:
Were these added in case you wanted to create a second 'AST' that just contained types?

Reply:
These were here originally because I copied the code from the parser's AST. This data is actually redundant now though, because we construct a type table with all of this information in name analysis.